ouroboros-consensus-shelley-test/test/Test/ThreadNet/RealTPraos.hs

```diff
@@ -2,124 +2,83 @@
 
 module Test.ThreadNet.RealTPraos (tests) where
 
-import           Control.Monad (replicateM)
 import           Data.List ((!!))
-import           Data.Word (Word64)
 
 import           Test.QuickCheck
 import           Test.Tasty
 import           Test.Tasty.QuickCheck
 
-import           Cardano.Slotting.Slot (EpochSize (..))
-
-import           Ouroboros.Consensus.Config.SecurityParam
 import           Ouroboros.Consensus.Ledger.SupportsMempool (extractTxs)
-import           Ouroboros.Consensus.Node.ProtocolInfo
 import           Ouroboros.Consensus.NodeId
-import           Ouroboros.Consensus.Util.Random
 
 import           Test.ThreadNet.General
 import           Test.ThreadNet.Infra.Shelley
+import           Test.ThreadNet.Rekeying
 
 import           Test.Util.Orphans.Arbitrary ()
 
-import qualified Shelley.Spec.Ledger.OCert as SL
-
 import           Ouroboros.Consensus.Shelley.Node
 
-import           Test.Consensus.Shelley.MockCrypto (TPraosMockCrypto)
 import           Test.ThreadNet.TxGen.Shelley
 import           Test.ThreadNet.Util.NodeJoinPlan (trivialNodeJoinPlan)
-import           Test.ThreadNet.Util.NodeRestarts (noRestarts)
-
-data TestSetup = TestSetup
-  { setupD          :: Double
-    -- ^ decentralization parameter
-  , setupK          :: SecurityParam
-  , setupTestConfig :: TestConfig
-  }
-  deriving (Show)
-
-instance Arbitrary TestSetup where
-  arbitrary = do
-    setupD <- (/10)         <$> choose   (1, 10)
-    setupK <- SecurityParam <$> elements [5, 10]
-
-    setupTestConfig <- arbitrary
-
-    pure TestSetup
-      { setupD
-      , setupK
-      , setupTestConfig
-      }
-
-  -- TODO shrink
 
 tests :: TestTree
 tests = testGroup "RealTPraos"
```

Comment:
This diff somewhat undoes my recent PR by moving the generator back into the `Infra.Shelley`. I assume that's so it can be shared with `ouroboros-consensus-cardano:test:test`. That seems very reasonable for now, but I anticipate that we wil eventually want separate test setup generators for those two.